test/integration/test_collections.js

```diff
@@ -66,6 +69,33 @@ module.exports = {
            expect(res.text).to.equal('test1,test2 collection(s) deleted');
            done();
          });
+    },
+
+    'test_collection_upload': function(done) {
+      const test = (ext, exts) => {
+
+        if (!(ext && exts)) {
+          done();
+        }
+
+        chai.request(SERVER_URL)
+        .post(`${PATH_PREFIX}/collections/upload`)
+        .attach('file', fs.readFileSync(`${__dirname}/fixture/import.${ext}`), `import.${ext}` )
+        .set('Authorization', `Bearer ${TOKEN}`)
+        .end((err, res) => {
+          expect(err).to.be.null;
+          expect(res).to.have.status(201);
+
+          mongodbClient.dropCollection('import', err => {
+            expect(err).to.be.null;
+
+            test(exts.shift(), exts);
+          });
+        });
+      };
+
+      const extensions = ['json','csv','bson'];
+      test(extensions.shift(), extensions);
```

Comment:
I just want to make sure we are ok with using the recursive method here. It's slightly different to how other tests are written.

The other option is using async and would look something like
```javascript
    'test_collection_upload': function(done) {
      const test = (ext, cb) => {
        chai.request(SERVER_URL)
          .post(`${PATH_PREFIX}/collections/upload`)
          .attach('file', fs.readFileSync(`${__dirname}/fixture/import.${ext}`), `import.${ext}` )
          .set('Authorization', `Bearer ${TOKEN}`)
          .end((err, res) => {
            expect(err).to.be.null;
            expect(res).to.have.status(201);
            mongodbClient.dropCollection('import', cb);
          });
      };

      async.eachSeries(['json','csv','bson'], function(ext, cb) {
        test(ext, cb);
      }, done);
    }
```

Thoughts @pb82 @rachael-oregan ?